internal/token/token.go

```diff
@@ -0,0 +1,53 @@
+// Package token provides functions to save and load tokens from disk.
+package token
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ubuntu/authd-oidc-brokers/internal/providers/info"
+	"golang.org/x/oauth2"
+)
+
+// AuthCachedInfo represents the token that will be saved on disk for offline authentication.
+type AuthCachedInfo struct {
```

Comment:
The more I look at it, the more I hate myself for using this name. Now that we have it in its package, can you think of a better name? I was thinking about using `CacheEntry` or just `CachedInfo`.

Reply:
My plan is to rename it to `Token` in a follow-up PR, see https://github.com/ubuntu/authd-oidc-brokers/pull/134#discussion_r1757031321